templates/deployment.yaml

```diff
@@ -68,189 +79,135 @@ spec:
             {{- end }}
-            {{- if .Values.adminUser.password }}
-            - name: ADMIN_USER_PWD
-              valueFrom:
-                secretKeyRef:
-                  key: ADMIN_USER_PWD
-                  name: {{ include "plausible-analytics.fullname" . }}
+            {{- if .Values.logFailedLoginAttempts }}
+            - name: LOG_FAILED_LOGIN_ATTEMPTS
+              value: {{ .Values.logFailedLoginAttempts | toString | quote }}
             {{- end }}
-            {{- if .Values.postgresql.url }}
             - name: DATABASE_URL
               valueFrom:
                 secretKeyRef:
                   key: DATABASE_URL
-                  name: {{ include "plausible-analytics.fullname" . }}
-            {{- end }}
-            - name: SECRET_KEY_BASE
-              valueFrom:
-                secretKeyRef:
-                  key: SECRET_KEY_BASE
-                  name: {{ include "plausible-analytics.fullname" . }}
-            {{- if .Values.clickhouse.url }}
+                  name: {{ include "plausible-analytics.secretName" . }}
             - name: CLICKHOUSE_DATABASE_URL
               valueFrom:
                 secretKeyRef:
                   key: CLICKHOUSE_DATABASE_URL
-                  name: {{ include "plausible-analytics.fullname" . }}
-            {{- end }}
-            {{- if .Values.clickhouse.auth.username }}
-            - name: CLICKHOUSE_USER
-              valueFrom:
-                secretKeyRef:
-                  key: CLICKHOUSE_USER
-                  name: {{ include "plausible-analytics.fullname" . }}
-            {{- end }}
-            {{- if .Values.clickhouse.auth.password }}
-            - name: CLICKHOUSE_PASSWORD
-              valueFrom:
-                secretKeyRef:
-                  key: CLICKHOUSE_PASSWORD
-                  name: {{ include "plausible-analytics.fullname" . }}
-            {{- end }}
-            {{- if .Values.smtp.enabled }}
-            {{- if .Values.smtp.mailer.emailAddress }}
+                  name: {{ include "plausible-analytics.secretName" . }}
+            {{- if .Values.mailer.enabled }}
+            {{- if .Values.mailer.email }}
             - name: MAILER_EMAIL
-              valueFrom:
-                secretKeyRef:
-                  key: MAILER_EMAIL
-                  name: {{ include "plausible-analytics.fullname" . }}
+              value: {{ .Values.mailer.email | toString | quote }}
             {{- end }}
-            {{- if .Values.smtp.adapter }}
+            {{- if .Values.mailer.adapter }}
             - name: MAILER_ADAPTER
-              valueFrom:
-                secretKeyRef:
-                  key: MAILER_ADAPTER
-                  name: {{ include "plausible-analytics.fullname" . }}
+              value: {{ .Values.mailer.adatper | toString | quote }}
             {{- end }}
-            {{- if .Values.smtp.host }}
+            {{- if eq .Values.mailer.adapter "Bamboo.SMTPAdapter" }}
```

Comment:
@mrdotb the same here, I think we should be able to set the SMTP_HOST_ADDR for any mailer adapters.